### PR TITLE
Update README to include info on default rules and conf file example

### PR DESCRIPTION
**Pull request by sealneaward**
Some improvements for the README.
I was getting lost with configuration files and found that not including the standard rules that are included in the Dockerfile conf file setup was preventing the default functionality to be carried over to custom conf files for the crs and modsecurity.

### CODE REVIEW

**Comment by jcmoraisjr** [comment]
Hi, thanks for the contribution! Sorry about taking so long to have a look.

Currently the default value for the config file comes from this assignment:

https://github.com/jcmoraisjr/modsecurity-spoa/blob/608e78d81c820d62b2d535dbfeef50cc2ec4526f/rootfs/start.sh#L29-L31

The second file has almost the same content when compared with your proposal:

```diff
/ # diff -u /etc/modsecurity/owasp-modsecurity-crs.conf proposal
--- /etc/modsecurity/owasp-modsecurity-crs.conf
+++ proposal
@@ -1,4 +1,4 @@
-Include /etc/modsecurity/owasp-modsecurity-crs/crs-setup.conf
+
 Include /etc/modsecurity/owasp-modsecurity-crs/rules/REQUEST-900-EXCLUSION-RULES-BEFORE-CRS.conf
 Include /etc/modsecurity/owasp-modsecurity-crs/rules/REQUEST-901-INITIALIZATION.conf
 Include /etc/modsecurity/owasp-modsecurity-crs/rules/REQUEST-903.9001-DRUPAL-EXCLUSION-RULES.conf
```

Maybe the readme could be improved to reference `/etc/modsecurity/modsecurity.conf` and `/etc/modsecurity/owasp-modsecurity-crs.conf` instead, already present in the image?

**Comment by sealneaward** [comment]
I'm fine with that, it was just confusing for a beginner to understand where to properly config the crs, because when you go to the [owasp crs repo](https://github.com/SpiderLabs/owasp-modsecurity-crs) it points someone like me to using the [example config file](https://github.com/SpiderLabs/owasp-modsecurity-crs/blob/v2/master/crs-setup.conf.example). When you use that example file as a replacement for the `/etc/modsecurity/owasp-modsecurity-crs.conf` file used in the image (without first inspecting what is in the `/etc/modsecurity/owasp-modsecurity-crs.conf` file), you would tend to overlook the default rules that your image pulls into the file.

That's all I'm trying to get accross. I'll update my branch.

**Comment by jcmoraisjr** [comment]
Thanks for the update, and sorry but I'm still a bit confused. My understanding so far is that:

- if no config file is added using `-f` or after the `--`, the default rules are applied
- if a customized config file is added using the parameters above, the defaults aren't used
    - in this case, in order to not loose the default rules, the user can add your own config file and also `/etc/modsecurity/modsecurity.conf` and `/etc/modsecurity/owasp-modsecurity-crs.conf` already provided in the image.

If this is true, why one need to create his own `owasp-modsecurity-crs.conf` with all that rules if a file with the same content is already provided in the image?

**Comment by sealneaward** [comment]
But if you are using your own config files, they would conflict with the default.
For example, I want to be able to change the default actions for `phase:1` and `phase:2` in `SecDefaultAction`.
If I just provide a file with 
```
SecDefaultAction "phase:1,log,auditlog,pass"
SecDefaultAction "phase:2,log,auditlog,pass"
```
It only uses that config file provided above and does not use the `/etc/modsecurity/modsecurity.conf` and `/etc/modsecurity/owasp-modsecurity-crs.conf` files already present in the image, so when you test your configuration, expecting default behaviour that includes the OWASP ruleset, it does nothing, because it does not include the default crs specified in `/etc/modsecurity/owasp-modsecurity-crs.conf`. For people who do not know what is inside the `/etc/modsecurity/owasp-modsecurity-crs.conf` file used by default, this can be confusing to understand why the ModSecurity agent is not doing anything when it is running.

**Comment by jcmoraisjr** [comment]
Completely agree about to clarify that the default configuration won't be used if a customization is made - the user can easily misconfigure the deployment. However I still didn't understand why the files should be copied if they are already in the container. Note also that if you mount /etc/modsecurity you'll hide lots of configuration files from OWASP ModSecurity CRS.

I've just added pr #9 improving the readme file. I just use the default configuration on our environments so, as I said before, I can be missing something here. Please let me know if #9 has the expected clarification.

**Comment by sealneaward** [comment]
In our implementation, we don't mount /etc/modsecurity, we chose to mount to a seperate directory.
That's something I can change in this PR if that's something you want to see changed.
It's possible that the user wants to use the default configuration files provided, but wants to modify them slightly.

For my use case, our team wanted to expose the default configuration files in our repository so that the client has insight into what is being configured for their WAF. If we don't copy the default configuration files and expose them in our product, then our clients wouldn't have a way of looking at the logs of the modsec agent pods in kubernetes and correlating a deny action respone with what was being configured as default.

If they have the ability to look at the deny action response, and determine they want to turn off all SQL injection rules for the WAF, they need to be able to go into all of the configuration files in our code for the WAF and comment out or delete this line in our configuration file.
```
Include /etc/modsecurity/owasp-modsecurity-crs/rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf
```

If you only expose a small portion of the configuration in the custom configuration option, then clients won't really know what to do in all different scenarios, unless they are already experts of modsecurity.

**Comment by jcmoraisjr** [comment]
Thank you for the time invested on this pr and to the component, and sorry again for taking so long to provide some feedback. I can see the value of your proposal, let's follow your approach but I'd suggest some few changes:

* Use something like "Customize the configuration files" and "Use default configuration", this seems to better contrast each other and clarifies what we steps are doing
* If you feel there is some space to something like "Add new configuration", this [commit](https://github.com/jcmoraisjr/modsecurity-spoa/commit/644b6d0b932362c4dca7f1f58e25e4cc5f2c4d9a) can be used as a baseline; I really don't know, maybe only the customized option should be enough
* Leave the code block only with code, move any comment to the line above, outside the code block
* The mounting point should be changed to not override the owasp conf files, or the cp instead to copy all the files to the local file system
* The copy file process can be optimized to the following since the container cannot be reused; of course container name can already exists, this is just a guideline anyway:

```
docker create --name modsec quay.io/jcmoraisjr/modsecurity-spoa
docker cp modsec:/etc/modsecurity/modsecurity.conf .
docker cp modsec:/etc/modsecurity/owasp-modsecurity-crs.conf .
docker rm modsec
```

**Comment by sealneaward** [comment]
@jcmoraisjr I updated from your review. Hopefully I got everything. If there are specific lines that you want to see the changes listed, that would be helpful to know by making the commends in the files changed section.

I also have some ideas about the integration with the HAProxy Ingress Controller that you maintain.
Both are ideas for additional containers to add to the daemonset definition you have in documentation.
Would that best be served as a feature request?

**Comment by jcmoraisjr** [comment]
Discovered a few more issues after trying the snippets:

* `unicode.mapping` is missing and it's used by `modsecurity.conf`;
* `crs-setup.conf` is also missing and copying it wouldn't work beucase it's referenced with an absolute path by `owasp-modsecurity-crs.conf`, so couldn't be properly customized.

So it seems to me that the better idea is to copy the whole modsec directory instead, and mount it in the same place - every configuration file could be safely changed. So copy would be something like this:

```
docker create --name modsec quay.io/jcmoraisjr/modsecurity-spoa
docker cp modsec:/etc/modsecurity .
docker rm modsec
```

and running something like this:

```
docker run -p 12345:12345 -v $PWD/modsecurity:/etc/modsecurity quay.io/jcmoraisjr/modsecurity-spoa -n 1
```

Note that you don't need to reference the configuration files, they are in the same place but optionally with a different content.

Regarding the controller a fr seems good, we've already some ideas to improve the controller's mod security usability and it'd be great to also fit your use case as well.

File: README.md [hunk 1]
```diff
@@ -22,8 +22,31 @@ $ docker run -p 12345:12345 quay.io/jcmoraisjr/modsecurity-spoa [options] [-- <c
...
+
+Download and customize the configuration files for either the [ModSecurity repository](https://github.com/SpiderLabs/ModSecurity/blob/v2/master/modsecurity.conf-recommended) or from [OWASP repository](https://github.com/SpiderLabs/owasp-modsecurity-crs/blob/v3.3/dev/crs-setup.conf.example).
+Use the copied files from the previous code section in your run command:
+```
+docker run -p 12345:12345 -v $PWD/modsecurity:/etc/modsecurity quay.io/jcmoraisjr/modsecurity-spoa -n 1
+```
+
+If you do not want to include the default configuration files and only use the configuration files (ex./ custom-config.conf) that you design, leave out the copied default configuration files from before in your run command:
+```
+docker run -p 12345:12345 -v $PWD:/etc/modsecurity/ quay.io/jcmoraisjr/modsecurity-spoa -n 1 -- /etc/modsecurity/custom-config.conf
```
Comment (jcmoraisjr) [inline README.md:44]: Missing `/modsecurity` after `$PWD`, something like this:

```suggestion
docker run -p 12345:12345 -v $PWD/modsecurity:/etc/modsecurity quay.io/jcmoraisjr/modsecurity-spoa -n 1 -- /etc/modsecurity/custom-config.conf
```

**Comment by jcmoraisjr** [comment]
Lgtm, thanks! Merging.

**Comment by sealneaward** [comment]
@jcmoraisjr I added my PR for a sidecar container to the docs repo. I also posted a question to the slack channel about the second container idea I had(not implemented yet).

**Pull request merged**